### PR TITLE
Move the `fuzz` directory into main workspace

### PR DESCRIPTION
**Pull request by alexcrichton**
This'll help us track dependencies and head off issues like #721

### CODE REVIEW

File: fuzz/Cargo.toml [hunk 1]
```diff
@@ -16,22 +16,22 @@ wasmtime-fuzzing = { path = "../crates/fuzzing", features = ["env_logger"] }
 wasmtime-jit = { path = "../crates/jit" }
 libfuzzer-sys = { git = "https://github.com/rust-fuzz/libfuzzer-sys.git" }
 
-# Prevent this from interfering with workspaces
-[workspace]
-members = ["."]
-
```
Comment (fitzgen) [inline fuzz/Cargo.toml]: I don't know why historically `cargo fuzz` did this, but maybe seems important? Or at least we should figure out why before we undo it.

Comment (fitzgen) [reply fuzz/Cargo.toml]: (Looking into the context now, btw)

Comment (fitzgen) [reply fuzz/Cargo.toml]: I can't find any historical reason for this other than cargo-fuzz's own CI set up, so I think we are good to go here...

Comment (alexcrichton) [reply fuzz/Cargo.toml]: Hm so thinking about this, one reason may be that libfuzzer-sys may need updates as nightlies progress, I'm not entirely sure how libfuzzer is compiled and all. If this becomes an issue we should revert and consider another route!

**Pull request merged**